forest/benchmarking/tomography.py

```diff
@@ -163,6 +167,98 @@ def linear_inv_state_estimate(results: List[ExperimentResult],
     rho = pinv(measurement_matrix) @ expectations
     return unvec(rho)
 
+def compressed_sensing_state_estimate(results: List[ExperimentResult],
+                              qubits: List[int]) -> np.ndarray:
+    """
+    Estimate a quantum state using compressed sensing
+
+    [FLAMMIA] Quantum Tomography via Compressed Sensing: Error Bounds, Sample Complexity, and Efficient Estimators
+           Steven T. Flammia et. al.
+           (2012).
+           https://arxiv.org/pdf/1205.2300.pdf
+
+    :param results: A tomographically complete list of results.
+    :param qubits: All qubits that were tomographized. This specifies the order in
+        which qubits will be kron'ed together.
+    :return: A point estimate of the quantum state rho.
+    """
+    pauli_num = len(results)
+    qubit_num = len(qubits)
+    d = 2 ** qubit_num
+    pauli_list = []
+    expectation_list = []
+    y = np.zeros((m,1))
+
+    for i in range(pauli_num):
+        #Convert the Pauli term into a tensor
+        r = results[i]
+        p_tensor = pauli2matrix(r.setting.out_operator, qubits)
+        e = r.expectation
+        #A[i] = e * scale_factor
+        pauli_list.append(p_tensor)
+        expectation_list.append(e)
+
```

Comment:
Perhaps add a comment  about the objective and constraint in terms of Eqn (3) and (34) in [FLAMMIA]